midterm_project_Nechita_Zaganeanu/shopping_list.py

```diff
@@ -0,0 +1,268 @@
+        return (string)
+
+#taken from https://stackoverflow.com/questions/19022868/how-to-make-dictionary-read-only-in-python
+class ReadOnlyDict(dict):
+    
+    def __readonly__(self, *args, **kwargs):
+        raise RuntimeError("Cannot modify Recipe")
+    
+    __setitem__ = __readonly__
+    __delitem__ = __readonly__
+    pop = __readonly__
+    popitem = __readonly__
+    clear = __readonly__
+    update = __readonly__
+    setdefault = __readonly__
+    del __readonly__
+    
+class Recipe(PrettyPrinter):
+    
+    def __init__(self, name, dictionary_of_ingredients):
+        self._name = name
+        self._dictionary_of_ingredients = ReadOnlyDict(dictionary_of_ingredients)
+    
+    @property
+    def name(self):
+        return self._name
+
+    def __getitem__(self, ingredient):
+        return self._dictionary_of_ingredients[ingredient]
+    
+    def __contains__(self, ingredient):
+        if ingredient in self._dictionary_of_ingredients:
+            return True
+        else:
+            return False
+    def get(self, key):
+        return self._dictionary_of_ingredients[key]
+    
+    def __eq__(self, other):
+        if isinstance(other, Recipe):
+            return self._dictionary_of_ingredients == other._dictionary_of_ingredients
+        return False
+    
+    def __ne__(self, other):
+        return not self.__eq__(other)
+       
+    def __iter__(self):
+        return iter(self._dictionary_of_ingredients)       
+        
+    def __len__(self):
+        return len(self._dictionary_of_ingredients)
+    
+    def items(self):
+        return self._dictionary_of_ingredients.items()
+    
+    def keys(self):
+        return self._dictionary_of_ingredients.keys()
+    
+    def values(self):
+        return self._dictionary_of_ingredients.values()
+       
+    def __str__(self):
+        return super().__str__()
+                    
+class RecipesBox():
+   
+    def __init__ (self, name, recipes_list):
+        self.name=name
+        self.recipes_list = recipes_list
+    
+    def __iter__(self):
+        return iter(self.recipes_list)
+    
+    def insert (self, index, recipe):
+        return self.recipes_list.insert(index, recipe)
+    
+    def __getitem__(self, index):
+        return self.recipes_list[index]
+    
+    def __setitem__(self, index, value):
+        self.recipes_list[index]=value
+    
+    def __delitem__(self, index):
+        del self.recipes_list[index]
+        
+    def __len__(self):
+        return len(self.recipes_list)
+    
+    def pick(self, recipe_to_extract=None):
+        if recipe_to_extract:
+            for recipe in self.recipes_list:
+                if recipe.name == recipe_to_extract:
+                    return recipe
+        else:
+            return random.choice(self.recipes_list)
+    
+    def add(self, recipe_to_add):
+        self.recipes_list.append(recipe_to_add)
+    
+    def delete(self, recipe_to_delete):
+        for recipe in self.recipes_list:

```

Comment:
si aici poti folosi abordarea cu `index()`

Reply:
acelasi lucru ca la metoda pick - recipe_to delete este atributul obiectului din lista, nu se afla in lista in sine